lib/rules/no-unused-vars.js

```diff
@@ -624,6 +645,19 @@ module.exports = {
                             if (config.varsIgnorePattern && config.varsIgnorePattern.test(def.name.name)) {
                                 continue;
                             }
+
+                            if (type === "Variable" && config.destructuredArrayIgnorePattern && config.destructuredArrayIgnorePattern.test(def.name.name)) {
+
+                                // Allow - `const [a, _b, c] = arr`, Disallow - `const [{ _a, foo }] = [obj];`
+                                if (def.node.id.type === "ArrayPattern" && def.name.parent.parent.type !== "ObjectPattern") {
+                                    continue;
+                                }
+
+                                // For cases like - `var { x: [_a, foo] } = obj;`
+                                if (def.node.id.type === "ObjectPattern" && def.name.parent.type === "ArrayPattern") {
+                                    continue;
+                                }
+                            }
```

Comment:
This would be a false positive on `_a`:

```js
/*eslint no-unused-vars: ["error", { "destructuredArrayIgnorePattern": "^_" }]*/

var [[a_, b]] = bar;
b;
```

I think the type of the top pattern (`def.node.id`) shouldn't matter, we can just check `def.name.parent.type === "ArrayPattern"`.

Reply:
fixed.